Cannot rebuild NixOS with system76 kernel modules

### Steps To Reproduce

On channels 22.05 and unstable, I cannot rebuild my configuration with `hardware.system76.kernel-modules.enable = true;`

### Build log
```
building the system configuration...
error: builder for '/nix/store/wvi23hy620m705x3mypz7bkdyw6vrxf0-system76-io-module-1.0.1-5.18.drv' failed with exit code 2;
       last 10 log lines:
       > In file included from /build/source/system76-io.c:38:
       > At top level:
       > /build/source/system76-io_dev.c:272:12: error: 'io_dev_suspend' defined but not used [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-function-Werror=unused-function8;;]
       >   272 | static int io_dev_suspend(struct io_dev * io_dev, u16 * value, int timeout) {
       >       |            ^~~~~~~~~~~~~~
       > cc1: all warnings being treated as errors
       > make[2]: *** [/nix/store/gkf4nhq306r54y7f9mfinmlpjppn04wc-linux-5.18-dev/lib/modules/5.18.0/source/scripts/Makefile.build:288: /build/source/system76-io.o] Error 1
       > make[1]: *** [/nix/store/gkf4nhq306r54y7f9mfinmlpjppn04wc-linux-5.18-dev/lib/modules/5.18.0/source/Makefile:1834: /build/source] Error 2
       > make[1]: Leaving directory '/nix/store/gkf4nhq306r54y7f9mfinmlpjppn04wc-linux-5.18-dev/lib/modules/5.18.0/build'
       > make: *** [Makefile:5: all] Error 2
       For full logs, run 'nix log /nix/store/wvi23hy620m705x3mypz7bkdyw6vrxf0-system76-io-module-1.0.1-5.18.drv'.
error: 1 dependencies of derivation '/nix/store/1i6zwf09hc889lb3a78zxpi1c8x8yy3c-hwdb.bin.drv' failed to build
error: 1 dependencies of derivation '/nix/store/3r7mwclv11ba9iz6v8fgai1lgbng6rmy-kernel-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/kav8bfxlaxlicwklig6v2aa99jb25anr-linux-5.18-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/xvimg9l7bwsciwypggm3jp53s4w7sr3s-unit-systemd-udevd.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/6ifcns4ac8zr3gkjsw5dakkrchf8xasi-nixos-system-oryx-22.05.20220530.a634c8f.drv' failed to build
```

### Notify maintainers
@khumba 

### Metadata
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.17.9, NixOS, 22.11 (Raccoon)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/os-specific/linux/firmware/system76-firmware/default.nix
+++ b/pkgs/os-specific/linux/firmware/system76-firmware/default.nix
@@ -2,13 +2,13 @@
 rustPlatform.buildRustPackage rec {
   pname = "system76-firmware";
   # Check Makefile when updating, make sure postInstall matches make install
-  version = "1.0.32";
+  version = "1.0.39";
 
   src = fetchFromGitHub {
     owner = "pop-os";
     repo = pname;
     rev = version;
-    sha256 = "sha256-EV7byqfoz0sAIkf/hgZh+m7D2+54DEnE4td4CTX6c4s=";
+    sha256 = "sha256-nPHBL73hmvW9z5SQjmfu+ozMXxUEajNQxNtE/V9QwZ0=";
   };
 
   nativeBuildInputs = [ pkg-config makeWrapper ];
@@ -17,7 +17,7 @@ rustPlatform.buildRustPackage rec {
 
   cargoBuildFlags = [ "--workspace" ];
 
-  cargoSha256 = "sha256-/bf5JQgyFN8TH7o2TWHX5sv/NkxBLJ495iVW8c9Vqis=";
+  cargoSha256 = "sha256-BrzicLj7FbUqRG1BgQIRqh801tRQpRZkHSiX3ekAYqc=";
 
   # Purposefully don't install systemd unit file, that's for NixOS
   postInstall = ''

